dev/fattest.simplicity/src/componenttest/topology/impl/LibertyServer.java

```diff
@@ -5958,6 +6110,69 @@ public Set<String> getInstalledAppNames(String... possiblyInstalledAppNames) thr
         return subset;
     }
 
+    private static final String INSTALL_FEATURE_MESSAGE_PREFIX = "CWWKF0012I: The server installed the following features:";
+
+    /**
+     * Returns a set of the features which were installed at runtime startup, based on the messages.log.
+     * We only look at the first occurrence of the message. This method does not look for subsequent feature changes.
+     *
+     * e.g.
+     * CWWKF0012I: The server installed the following features: [bells-1.0, cdi-4.0, componenttest-2.0, concurrent-3.0, jndi-1.0, mpConfig-3.1, mpContextPropagation-1.3,
+     * mpFaultTolerance-4.0, servlet-6.0, timedexit-1.0].
+     *
+     * @return           a set of the features installed at runtime
+     * @throws Exception
+     */
+    public Set<String> getInstalledFeatures() throws Exception {
+        Set<String> installedFeatures = new HashSet<>();
+
+        for (String line : findStringsInLogs(INSTALL_FEATURE_MESSAGE_PREFIX)) {
+            installedFeatures.addAll(getInstalledFeaturesFromLogMessage(line));
+            break; //only look at the first message
+        }
```

Comment:
This seems a little questionable, but I'm not sure how many tests it would affect.

I guess if tests are deliberately changing their features, they're less likely to be relying on `FeatureReplacementAction`.

Reply:
I think this is ok